lib/searchable_test.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2020-2022 CYBERCRYPT
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package encryptonize
+
+import (
+	"encryptonize/crypto"
+	"testing"
+)
+
+func TestNewSearchable(t *testing.T) {
+	searchable := NewSearchable()
+	if len(searchable.Mapping) != 0 {
+		t.Fatal("Index non-empty at initialization.")
+	}
+}
+
+func TestAdd(t *testing.T) {
+	searchable := NewSearchable()
+
+	rand := &crypto.NativeRandom{}
+	masterKey, err := rand.GetBytes(32)
+	if err != nil {
+		t.Fatalf("Random failed: %v", err)
+	}
+
+	keyword := "first keyword"
+	id := "first id"
+
+	err = searchable.Add(masterKey, keyword, id)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(searchable.Mapping) != 1 {
+		t.Fatal("Keyword/id pair not correctly added to mapping")
+	}
+}
+
+func TestAddMultiple(t *testing.T) {
+	searchable := NewSearchable()
+
+	rand := &crypto.NativeRandom{}
+	masterKey, err := rand.GetBytes(32)
+	if err != nil {
+		t.Fatalf("Random failed: %v", err)
+	}
+
+	keywords := [2]string{"first keyword", "second keyword"}
+	ids := [2]string{"first id", "second id"}
+
+	for i := 0; i < len(keywords); i++ {
+		for j := 0; j < len(ids); j++ {
+			err = searchable.Add(masterKey, keywords[i], ids[j])
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	if len(searchable.Mapping) != 4 {
```

Comment:
To make life easier in the future. 
```suggestion
	if len(searchable.Mapping) != len(keywords)*len(ids) {
```